frontend/wallet/src/main/IpcLinkMain.re

```diff
@@ -0,0 +1,74 @@
+open Tc;
+
+module GraphqlIpcMain = BsElectron.IpcMain.MakeIpcMain(GraphqlLinkMessages);
+
+let sendStringified = (event, id, stringified, errorMsg) => {
+  let response: Tc.Result.t(string, string) =
+    switch (stringified) {
+    | Some(responseStr) => Ok(responseStr)
+    | None => Error(errorMsg)
+    };
+
+  event##sender##send(
+    GraphqlLinkMessages.message,
+    Js.Json.stringify(
+      [@warning "-20"]
+      BsElectron.Json.toValidJson(`Pipe_graphql_response((id, response))),
+    ),
+  );
+};
+
+let start = apolloClient =>
+  GraphqlIpcMain.on((. event, m) =>
+    switch (m) {
+    | `Pipe_graphql_request(id, requestStr) =>
+      // These are the same thing
+      let r: ApolloClient.queryObj = Obj.magic(Js.Json.parseExn(requestStr));
+      let query =
+        Task.liftPromise(() =>
```

Comment:
could you make a `Task.liftErrorPromise` (or something with a better name), that takes a `unit => Promise.t('a)` like this one, but does the catching for you and sticks it in the `'x` part of the task.

So that way `apolloClient##query(r)` could be turned into a `Task.t(Js.Exn.t, whateveritis)` instead of a `Task.t('x, [ | `Ok(whateveritis) | `Error(Js.Exn.t)])`.

Then later you can `Task.attempt` the task and you get a Result.t out instead of the polymorphic variant.